src/credentials/examples/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -257,6 +257,14 @@ AttestationVerificationResult DefaultDACVerifier::VerifyAttestationInformation(c
     VerifyOrReturnError(GetProductAttestationAuthorityCert(akid, paa) == CHIP_NO_ERROR,
                         AttestationVerificationResult::kPaaNotFound);
 
+    VerifyOrReturnError(IsCertificateTimestampValid(CertificateValidityType::kNotBefore, dacCertDerBuffer, paiCertDerBuffer) == CHIP_NO_ERROR &&
```

Comment:
Do we also need to verify that DAC validity time is also valid at the current time?

Reply:
I think we have to, but I think we should also handle the notAfter infinity value.

Reply:
This is the original requirement - "We need to apply checks on PAI/PAA based on DAC time, not current time". If we are going to verify the DAC validity time against current time, then what is the point of validating PAI/PAA based on DAC time?
I propose that we validate if DAC, PAI and PAA are valid at the current time. Do you agree?

Reply:
No, I do not agree. The spec says the following:

> The validity period of a DAC is determined by the vendor and MAY be set to the maximum allowed value of 99991231235959Z `GeneralizedTime` to indicate that the DAC has no well-defined expiration date.

...

> * The PAA SHALL be validated for presence in the Commissioner's trusted root store, which SHOULD include at least the set of globally trusted PAA certificates present in the <<ref_DistributedComplianceLedger, Distributed Compliance Ledger>> at the issuing timestamp (`notBefore`) of the DAC.
> * The DAC certificate chain SHALL be validated using the <<ref_PUBCert, `Crypto_VerifyChainDER()`>> function, taking into account the mandatory presence of the PAI and of the PAA.
>   ** Chain validation SHALL be performed with respect to the `notBefore` timestamp of the DAC to ensure that the DAC was valid when it was issued. This way of validating is abided by the <<ref_PUBCert, `Crypto_VerifyChainDER()`>> function.
>   ** Chain validation SHALL include revocation checks of the DAC, PAI and PAA, based on the Commissioner's best understanding of revoked entities.
> 
...
> The Validity validation for the root and optional intermediate certificates is performed against the `notBefore` timestamp of the end-entity (leaf certificate) used as value for the current time.

I think the spec may be a bit vague about DAC expiry itself, in that in theory the DAC may expire if you don't set a notAfter of 99991231235959Z.

However, the PAA/PAI validity need to be checked only w.r.t the leaf certificate notBefore time.

Reply:
I tend to agree with @vijs. All we need to validate is that all certificates in the certificate chain are valid at the current time.

I don't know why we need to verify that DAC validity period is within validity period of PAI. 

@tcarmelveilleux what do you think?

Reply:
Fixed in 4ef7efd. However, I've temporarily commented the time validations as we need to discuss the PAA/PAI re-issue use case.